### PR TITLE
Added mocking data and mocking integration tests for azure

### LINKED ISSUE #1531: Move azure integration test to mock provider
Move azure to mock provider, reduce the testing time

### PR DESCRIPTION
**Pull request by JunqiZhang0**
Added mocking data and mocking integration tests for azure. It also differentiated the present data and absent data. All other providers should do the same. fix #1531

### CODE REVIEW

File: linchpin/provision/roles/azure/molecule/docker/molecule.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  options:
+    config-file: tests/yamllint.yml
+platforms:
+  - name: azure-f31
+    image: lp_fedora31:latest
+    pull: false
+  - name: azure-f30
+    image: lp_fedora30
+    pull: false
+    ## Remove 'image' above and uncomment the following to run systemd in Docker
```
Comment (14rcole) [inline linchpin/provision/roles/azure/molecule/docker/molecule.yml]: Can you remove the commented lines here?  I have to go back and do it for AWS too

**Comment by 14rcole** [comment]
Junqi, what do you mean by differentiating present and absent data?  Isn't the data generated by provisioning (present) the data used by teardown (absent)?

**Comment by JunqiZhang0** [comment]
@14rcole No, they are using different data, I added code for SK's collector to collect both present data and absent data. Do you think it's necessary?  I personally think it's not that necessary to test the Teardown part because if the provision part works fine then the teardown should also work fine. But @samvarankashyap  said for consistency, we need to provide both data, I agree. What's your opinion?

File: linchpin/provision/roles/azure/molecule/docker/molecule.yml (hunk 1 above)
Comment (JunqiZhang0) [reply linchpin/provision/roles/azure/molecule/docker/molecule.yml]: yes

**Comment by JunqiZhang0** [comment]
[test]

**Comment by JunqiZhang0** [comment]
[test]

**Comment by JunqiZhang0** [comment]
[test]

**Comment by 14rcole** [comment]
Do you mean that the provider returns different data on teardown?

**Comment by JunqiZhang0** [comment]
@14rcole yes

**Comment by JunqiZhang0** [comment]
[test]

**Comment by JunqiZhang0** [comment]
[test]

**Comment by JunqiZhang0** [comment]
[test]

**Comment by JunqiZhang0** [comment]
[TEST]

**Comment by JunqiZhang0** [comment]
[test]

**Comment by JunqiZhang0** [comment]
[merge]

**Pull request merged**